core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksStore.java

```diff
@@ -217,11 +234,30 @@ public synchronized void writeToCheckpoint(OutputStream output)
   public synchronized void restoreFromCheckpoint(CheckpointInputStream input) throws IOException {
     LOG.info("Restoring rocksdb from checkpoint");
     long startNano = System.nanoTime();
-    Preconditions.checkState(input.getType() == CheckpointType.ROCKS,
+    Preconditions.checkState(input.getType() == CheckpointType.ROCKS
+        || input.getType() == CheckpointType.ROCKS_ZIP,
         "Unexpected checkpoint type in RocksStore: " + input.getType());
     stopDb();
     FileUtils.deletePathRecursively(mDbPath);
-    TarUtils.readTarGz(Paths.get(mDbPath), input);
+
+    if (input.getType() == CheckpointType.ROCKS_ZIP) {
+      List<String> tmpDirs = Configuration.getList(PropertyKey.TMP_DIRS);
+      String tmpZipFilePath = new File(tmpDirs.get(0), "alluxioRockStore-" + UUID.randomUUID())
+          .getPath();
+      try {
+        try (FileOutputStream fos = new FileOutputStream(tmpZipFilePath)) {
+          IOUtils.copy(input, fos);
+        }
+
+        ParallelZipUtils.decompress(Paths.get(mDbPath), tmpZipFilePath,
+            mParallelBackupPoolSize);
+      } finally {
+        FileUtils.deletePathRecursively(tmpZipFilePath);
+      }
```

Comment:
I figure the temp files can be left for analysis if something wrong happens. WDYT @tcrain ?
```suggestion
        boolean complete = false;
        try (FileOutputStream fos = new FileOutputStream(tmpZipFilePath)) {
          IOUtils.copy(input, fos);
        }

        ParallelZipUtils.decompress(Paths.get(mDbPath), tmpZipFilePath,
            mParallelBackupPoolSize);
        complete = true;
      } finally {
        if (complete) {
          FileUtils.deletePathRecursively(tmpZipFilePath);
        } else {
          LOG.warn("Failed to decompress checkpoint from {} to {}, input.getPathBlah(), tmpZipFilePath);
          // Leave the temp files for further investigation
        }
      }
```

Reply:
@adol001 see above ^